pkg/controller/controller.go

```diff
@@ -599,12 +599,17 @@ func ProcessUpdate(ctx context.Context, obj interface{}, resources []resource.In
 	return nil
 }
 
-func hasAnnotation(m map[string]string, k string, v string) bool {
-	if m == nil {
+func hasAnnotation(annotations map[string]string, targetKey string, targetValue string) bool {
+	if annotations == nil {
 		return false
 	}
 
-	return m[k] == v
+	actualValue, ok := annotations[targetKey]
+	if !ok {
+		return false
+	}
```

Comment:
It seems to me that it fixes only the case when `hasAnnotation(m, "non-existent", "") is called. Was this the issue?

Reply:
Exactly. I added the unit tests before changing the code and confirmed that this was the only case that was failing.